apps/designer/app/shared/router-utils/canvas-request-params.ts

```diff
@@ -0,0 +1,58 @@
+import config from "~/config";
+
+export type CanvasRouteMode = "edit" | "preview" | "published";
+
+const modes = ["edit", "preview", "published"] as CanvasRouteMode[];
+
+type ProjectIdentifier = { type: "id" | "domain"; value: string };
+
+export const getCanvasRequestParams = (request: {
+  url: string;
+  headers: { get: (name: string) => string | null };
+}):
+  | {
+      projectIdObject: ProjectIdentifier;
+      mode: CanvasRouteMode;
+      pathname: string;
+    }
+  | undefined => {
+  const url = new URL(request.url);
+
+  const pathname = url.pathname;
+
+  if (Object.values(config).some((path) => pathname.startsWith(path))) {
```

Comment:
this would return true for non-canvas paths too, e.g. dashboardPath, is that what you wanted?

Reply:
All paths that are remaining in `config` are not canvas paths. This line is correct. It checks for example that you're trying to access `/dashboard` and `getCanvasRequestParams` returns `undefined` in this case

Reply:
but there is no guarantee that config will only have non-canvas paths, right? also that config is not just for paths, it could get something else added to it ... later and the person who adds to config won't know this logic depends on it in such a way

Reply:
I've removed this check from `getCanvasRequestParams` now, and added an alternative check in `apps/designer/app/entry.server.tsx`. The new check doesn't use `config`.